crates/bevy_ecs/src/query/mod.rs

```diff
@@ -51,6 +53,145 @@ mod tests {
         assert_eq!(values, vec![&B(3)]);
     }
 
+    #[test]
+    fn query_filtered_len() {
```

Comment:
This test is a little verbose. Could you refactor out a helper method for the assertions?

Reply:
Okay sure.

Reply:
I tried refactoring the asserts to a helper function, and unfortunately when the test fails it does not point the line to the invocation of the function, but rather the first line of the helper function. This makes debugging harder.

Reply:
Also, turning the helper function to a macro solves the problem of pointing at the offending line, but it does not point at which specific assert is wrong. Ergo, you cannot tell the difference whether `.size_hint() != n` or `.len() != n`, or both.